File: apps/admin-v2/src/application/use-cases/settings/update-settings.use-case.ts
import type { TenantRepository } from '@/domain/repositories/tenant.repository';
import type { TenantSettings } from '@/domain/entities/tenant';

export class UpdateSettingsUseCase {
  constructor(private repo: TenantRepository) {}

  execute(data: Partial<TenantSettings>) {
    return this.repo.updateSettings(data);
  }
}

File: apps/admin_mobile/lib/application/use_cases/settings/update_settings_use_case.dart
import '../../../domain/repositories/tenant_repository.dart';

class UpdateSettingsUseCase {
  final TenantRepository _repo;
  UpdateSettingsUseCase(this._repo);

  Future<Map<String, dynamic>> call(Map<String, dynamic> data) =>
      _repo.updateSettings(data);
}
